### DISCUSSION TITLE
[Course-1, Lesson-4, Concept-8,9] Differential Privacy: What guarantees does DP gives about the accuracy of the outputs if the number of inputs is smaller than the range of random sampling?

### DISCUSSION THREAD

**Discussion by imflash217** [OpenMined/courses#152] [Course: Our Privacy Opportunity] [answered]
Hi,
As discussed by @iamtrask in L4/C-8,9; one of the benefits of DP is that the reconstruction of the inputs from the outputs remain noisy. One of the reason is the addition of randomly sampled values that gets added to the inputs. 
Referring to the discussed average-age-survey example, suppose that `num_participants_ << range_of_random_number_generator` (`10` participants & `np.random.randint(-100,100)` range i.e. `10<<200`); then because we haven't sampled very large number of random values the `output` will be extremely noisy (as the random number won't add to zero due to low sampling). 

Now, if we are running a product/service (like medical diagnosis or chemical-research) where the induced error could potentially lead to very wrong predictions due to undersampling of noise or a product where the accuracy of the outputs is of prime importance then how do we integrate DP in our algorithms in the first place to avoid this noise.

Please let me know if my understanding is wrong.
Great course. Thanks a lot. 🙂
Vinay

**Comment by em-blue** [comment, marked as answer]
hey @imflash217 great question! Course 2 will cover this very well :-) 
The short answer is: there are a lot of different methods of using differential privacy!
@ivyclare wrote a great blog post on one of these methods for beginners a few months ago https://blog.openmined.org/maintaining-privacy-in-medical-data-with-differential-privacy/

**Reply by imflash217** [reply]
Awesome. Looking forward to Course-2. 😃

**Reply by snaik2016** [reply]
Thanks for very interesting blog for PATE analysis. Usually a student is trained on soft labels generated from the ensemble. Is there any difference if we train the student on soft labels instead of hard label determined by voting?
